src/System/src/Barrier.cpp

```diff
@@ -20,9 +21,11 @@ Barrier::Barrier(const std::size_t counter)
 
 void Barrier::wait()
```

Comment:
I add it here because I don't know where to add it ;)

Since in the end the `Barrier` class needs to be shared among different tread I will force the user to use shared pointer to do this you can:
1. make the Barrier constructor private
2. create a static method called `std::shared_ptr<Barrier> Barrier::create(const std::size_t counter)`  implemented as follows
   ```cpp
   std::shared_ptr<Barrier> Barrier::create(const std::size_t counter)
   {
       return std::make_shared<Barrier>(counter);
   }
   ```

Reply:
Thanks! Applied with f882832c061cd4c44a27dc11536c5546e4b11f8f